### PR TITLE
docs(platform/gitlab): Mention DOCKER_REGISTRY_* CI variables for authentication

### PR DESCRIPTION
**Pull request by mbrodala**
See https://github.com/renovatebot/renovate/issues/17940

## Changes

No change, only documentation of current state.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository
- [x] Actual usage

### CODE REVIEW

**Review by rarkins — DISMISSED** [review]

File: lib/modules/platform/gitlab/readme.md [hunk 1]
```diff
@@ -41,6 +41,8 @@ Remember to set `platform=gitlab` somewhere in your Renovate config file.
 If you're using a private [GitLab container registry](https://docs.gitlab.com/ee/user/packages/container_registry/), you must:
 
 - Set the `RENOVATE_HOST_RULES` CI variable to `[{"matchHost": "${CI_REGISTRY}","username": "${GITLAB_USER_NAME}","password": "${RENOVATE_TOKEN}", "hostType": "docker"}]`.
+
+  Alternatively you can set the CI variables `DOCKER_REGISTRY_GITLAB_COM_USERNAME=${GITLAB_USER_NAME}` and `DOCKER_REGISTRY_GITLAB_COM_PASSWORD=${RENOVATE_TOKEN}`. (See [issue #17940](https://github.com/renovatebot/renovate/issues/17940))
```
Comment (viceice) [inline lib/modules/platform/gitlab/readme.md:45]: those don't work out of the box,you need to enable detect host rules from env

https://docs.renovatebot.com/self-hosted-configuration/#detecthostrulesfromenv

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/modules/platform/gitlab/readme.md (hunk 1 above)
Comment (mbrodala) [reply lib/modules/platform/gitlab/readme.md]: Added this now, thanks. Not sure about the relative link however.

File: lib/modules/platform/gitlab/readme.md [hunk 2]
```diff
@@ -41,6 +41,8 @@ Remember to set `platform=gitlab` somewhere in your Renovate config file.
 If you're using a private [GitLab container registry](https://docs.gitlab.com/ee/user/packages/container_registry/), you must:
 
 - Set the `RENOVATE_HOST_RULES` CI variable to `[{"matchHost": "${CI_REGISTRY}","username": "${GITLAB_USER_NAME}","password": "${RENOVATE_TOKEN}", "hostType": "docker"}]`.
+
+  Alternatively, if [`detectHostRulesFromEnv`](../../../../docs/usage/self-hosted-configuration.md#detecthostrulesfromenv) is enabled, you can set the CI variables `DOCKER_REGISTRY_GITLAB_COM_USERNAME=${GITLAB_USER_NAME}` and `DOCKER_REGISTRY_GITLAB_COM_PASSWORD=${RENOVATE_TOKEN}`. (See [issue #17940](https://github.com/renovatebot/renovate/issues/17940))
```
Comment (viceice) [inline lib/modules/platform/gitlab/readme.md:45]:
```suggestion
  Alternatively, if [`detectHostRulesFromEnv`](../../../../docs/usage/self-hosted-configuration.md#detecthostrulesfromenv) is enabled, you can set the CI variables `DOCKER_REGISTRY_GITLAB_COM_USERNAME=${GITLAB_USER_NAME}` and `DOCKER_REGISTRY_GITLAB_COM_PASSWORD=${RENOVATE_TOKEN}`.
```

I would not link to an issue here, link to the correct docs section instead.

Comment (mbrodala) [reply lib/modules/platform/gitlab/readme.md:45]: Yeah, probably makes sense. But there is no section in the docs yet which explicitly mention and showcase these environment variables AFAIS. So I guess for now the link to `detectHostRulesFromEnv` will do.

**Review by rarkins — DISMISSED** [review]

**Comment by rarkins** [comment]
```
          error: 'File not found: ../../../../docs/usage/self-hosted-configuration.md#detecthostrulesfromenv in modules/platform/gitlab/index.md -> usage/self-hosted-configuration.md'
```

Not sure if we just need to fiddle one more or less `../` or if we need a different approach to linking

**Comment by mbrodala** [comment]
For now linking via `/basically/absolute/paths` would probably help avoiding knots in brains. ;-)

But yeah, in the long term the docs may need a more reliable solution for interlinking.

**Comment by rarkins** [comment]
@HonkingGoose any advice on the linking part?

**Comment by HonkingGoose** [comment]
> For now linking via `/basically/absolute/paths` would probably help avoiding knots in brains. ;-)

I also dislike the long `../.../../../../finally-we-have-the-file.md` style paths. I don't know if using a absolute link is better here, or what the drawbacks are later...

As I understand it, we generally prefer relative links. Relative links allow the docs generator (`mkdocs`) to check if the file we link to is present in the build, I think.

> @HonkingGoose any advice on the linking part?

I'll ping @viceice, they know more than me about linking stuff. 😄

File: lib/modules/platform/gitlab/readme.md [hunk 3]
```diff
@@ -41,6 +41,9 @@ Remember to set `platform=gitlab` somewhere in your Renovate config file.
 If you're using a private [GitLab container registry](https://docs.gitlab.com/ee/user/packages/container_registry/), you must:
 
 - Set the `RENOVATE_HOST_RULES` CI variable to `[{"matchHost": "${CI_REGISTRY}","username": "${GITLAB_USER_NAME}","password": "${RENOVATE_TOKEN}", "hostType": "docker"}]`.
+
+  Alternatively, if [`detectHostRulesFromEnv`](../../../../docs/usage/self-hosted-configuration.md#detecthostrulesfromenv) is enabled, you can set the CI variables `DOCKER_REGISTRY_GITLAB_COM_USERNAME=${GITLAB_USER_NAME}` and `DOCKER_REGISTRY_GITLAB_COM_PASSWORD=${RENOVATE_TOKEN}`.
```
Comment (viceice) [inline lib/modules/platform/gitlab/readme.md:45]:
```suggestion
  Alternatively, if [`detectHostRulesFromEnv`](../../../../self-hosted-configuration.md#detecthostrulesfromenv) is enabled, you can set the CI variables `DOCKER_REGISTRY_GITLAB_COM_USERNAME=${GITLAB_USER_NAME}` and `DOCKER_REGISTRY_GITLAB_COM_PASSWORD=${RENOVATE_TOKEN}`.
```

**Review by viceice — COMMENTED** [review]
the modules readme's are copied relative to the docs folder on docs build, so docs folder is the root.

absolute docs urls would also fail the CI tests.

File: lib/modules/platform/gitlab/readme.md [hunk 4]
```diff
@@ -41,6 +41,9 @@ Remember to set `platform=gitlab` somewhere in your Renovate config file.
 If you're using a private [GitLab container registry](https://docs.gitlab.com/ee/user/packages/container_registry/), you must:
 
 - Set the `RENOVATE_HOST_RULES` CI variable to `[{"matchHost": "${CI_REGISTRY}","username": "${GITLAB_USER_NAME}","password": "${RENOVATE_TOKEN}", "hostType": "docker"}]`.
+
+  Alternatively, if [`detectHostRulesFromEnv`](../../../../self-hosted-configuration.md#detecthostrulesfromenv) is enabled, you can set the CI variables `DOCKER_REGISTRY_GITLAB_COM_USERNAME=${GITLAB_USER_NAME}` and `DOCKER_REGISTRY_GITLAB_COM_PASSWORD=${RENOVATE_TOKEN}`.
```
Comment (viceice) [inline lib/modules/platform/gitlab/readme.md:45]:
```suggestion
  Alternatively, if [`detectHostRulesFromEnv`](../../../self-hosted-configuration.md#detecthostrulesfromenv) is enabled, you can set the CI variables `DOCKER_REGISTRY_GITLAB_COM_USERNAME=${GITLAB_USER_NAME}` and `DOCKER_REGISTRY_GITLAB_COM_PASSWORD=${RENOVATE_TOKEN}`.
```

**Review by viceice — APPROVED** [review]

**Pull request merged**